src/NServiceBus.Transport.RabbitMQ/Connection/ConnectionFactory.cs

```diff
@@ -42,6 +42,11 @@ public ConnectionFactory(ConnectionConfiguration connectionConfiguration, X509Ce
             connectionFactory.Ssl.Version = SslProtocols.Tls12;
             connectionFactory.Ssl.Enabled = connectionConfiguration.UseTls;
 
+            if (useExternalAuthMechanism)
+            {
+                connectionFactory.AuthMechanisms = new[] { new ExternalMechanismFactory() };
```

Comment:
By default, `AuthMechanisms` contains a single `RabbitMQ.Client.PlainMechanismFactory`. Should we be adding to the collection here, rather than replacing it?

Reply:
It needs to replace it because the client will choose the first one in the collection that matches any of the auth mechanisms presented by the server.

SO, if the server is configured to allow both PLAIN and EXTERNAL, then if we just added to the collection, we'd always choose PLAIN, making it impossible to use the newly added EXTERNAL option.

Reply:
Thanks @bording. That's what I suspected (as did @WojcikMike) but I wanted to get confirmation. If you rebase this, I'll be happy approve and merge, assuming we're happy for this feature to be added during the beta phase (I don't see why not).